### DISCUSSION TITLE
[FR] Offer different GraphQL IDEs such as GraphQL Playground as an option

### DISCUSSION THREAD

**Discussion by khalwat** [craftcms/cms#8731] [Ideas]
Tools like Gatsby allow you to choose from several different default GraphQL IDEs via the `GATSBY_GRAPHQL_IDE` environment variable: https://www.gatsbyjs.org/docs/using-graphql-playground/

It'd be nice if a similar thing were possible with Craft CMS, in terms of being agnostic about the IDE of choice, and letting people use GraphiQL or GraphQL Playground as they see fit.

Related: https://github.com/prisma/graphql-playground

**Comment by BenParizek** [comment]
I would love to see GraphQL Playground as an option or the only option in the CP. I haven't had much luck with GraphiQL.

**Comment by narration-sd** [comment]
Seems not a bad idea at all, @BenParizek Ben -- @andris-sevcenko has been improving Gql's, but he's got several heavy responsibilities going, and this would give him respite from this facet.

Playground is in or a config option in other frameworks, and optioning it for Gql might be a nice way of taking the load off, plus offering an alternate that may be familiar to some.

In the meantime, the latest version (7.0.1) of [Insomnia](https://insomnia.rest/) is quite good and has a lot to offer besides the missing features. I like it for the memory of queries that doesn't get lost with browser cache clears, that you can export a myriad list of queries, and that it's a truth source for all matters of actually using the Gql or other api -- such as authorizations, which people have trouble with.

**Comment by mylesthedev** [comment]
+1
